### PR TITLE
Major cleanup refactor of www/

### PR DESCRIPTION
**Pull request by ingydotnet**
No build/ source subdir anymore

main/ source files are built live by Jekyll

The build dirs are:

* build/ - Jekyll output area for make serve
* site/ - gh-pages area for make publish
* html/ - html output for make html

Much less Makefile cruft
Builds faster

### CODE REVIEW

File: www/main/spec.scss [hunk 1]
```diff
@@ -1,7 +1,9 @@
+---
+---
 $theme-color: #606c71;
 
 body.status-draft {
-  background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' version='1.1' height='200px' width='200px'><text transform='translate(20, 200) rotate(-45)' fill='rgb(245,245,245)' font-size='60'>DRAFT</text></svg>");
+  background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' version='1.1' height='200px' width='200px'><text transform='translate(20, 200) rotate(-45)' fill='rgb(245,245,245)' font-size='60'>RAFT</text></svg>");
```
Comment (Thom1729) [inline www/main/spec.scss:6]: Unintended change, I presume.

File: www/external/ext.mk [hunk 2]
```diff
@@ -3,7 +3,6 @@ EXT_DIRS := \
     ext/eatme \
     ext/yaml-reference-parser \
     ext/npm-yaml \
-    ext/yaml-spec-12.html \
```
Comment (Thom1729) [inline www/external/ext.mk:6]: Was this unneeded before?

File: www/config.yaml [hunk 3]
```diff
@@ -3,15 +3,22 @@
...
 
-destination: /host/www/build/site
+destination: /host/www/build
 
 kramdown:
   parse_block_html: true
 
 theme: jekyll-theme-cayman
 
+layouts_dir: layouts
```
Comment (Thom1729) [inline www/config.yaml:13]: Any particular reason not to just use the standard names, e.g. `_layouts`?

Comment (ingydotnet) [reply www/config.yaml:13]: I think the defaults are for lazy people. I'm not lazy.

File: www/main/spec.scss (hunk 1 above)
Comment (ingydotnet) [reply www/main/spec.scss:6]: Yeah I was testing live updates. Thanks.

File: www/external/ext.mk (hunk 2 above)
Comment (ingydotnet) [reply www/external/ext.mk:6]: Correct. The file is committed here https://github.com/yaml/yaml-spec/blob/jekyll-live-update/www/external/spec-2009/yaml-spec-12.html

**Review by Thom1729 — APPROVED** [review]

File: www/Makefile [hunk 4]
```diff
@@ -175,12 +175,13 @@ clean:
 	rm -fr $(BUILD) $(SITE) $(HTML)
 	rm -fr $(FILES)
 	find . -empty -type d | xargs rm -fr
+	find . -empty -type d | xargs rm -fr
```
Comment (Thom1729) [inline www/Makefile:178]: Extra line?

File: www/main/playground/devel.html [hunk 5]
```diff
@@ -72,5 +72,4 @@
 - one
 - two
 - null
-```
```
Comment (Thom1729) [inline www/main/playground/devel.html:75]: Was this deliberate?

File: www/Makefile (hunk 4 above)
Comment (ingydotnet) [reply www/Makefile:178]: No. this is supposed to delete all empty dirs. But a dir containing an empty dir is not empty

File: www/main/playground/devel.html (hunk 5 above)
Comment (ingydotnet) [reply www/main/playground/devel.html:75]: That was a botch from `s/```/<pre>/` conversion.
Now `/main/playground/devel/` works again

File: www/Makefile (hunk 4 above)
Comment (ingydotnet) [reply www/Makefile:179]: Added a comment that get's printed during `make clean` as well

**Review by Thom1729 — APPROVED** [review]

File: www/Makefile (hunk 4 above)
Comment (perlpunk) [reply www/Makefile:179]: `find . -depth -empty -type d -delete`?

Comment (perlpunk) [reply www/Makefile:179]: #163

**Pull request merged**